### PR TITLE
create basic react structure done

### PR DESCRIPTION
**Pull request by daniahih**
create basic react structure done , i install the error boundry and try to test it in small example and now i comment it ,

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -10,8 +10,14 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
+    "i": "^0.3.7",
```
Comment (yishain11) [inline client/package.json:14]: what is this? did you add this by mistake?
if yes - please remove it

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/package.json (hunk 1 above)
Comment (daniahih) [reply client/package.json:14]: yes it by mistake , i will remove it

File: client/src/App.jsx [hunk 2]
```diff
@@ -1,34 +1,43 @@
...
+  },
+  {
+    path: "/song",
+    element: <SongPage />,
+  },
+  {
+    path: "/Artist",
+    element: <ArtistPage />,
+  },
+]);
```
Comment (yishain11) [inline client/src/App.jsx:20]: good job adding the basic boilerplate components

File: client/src/main.jsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 import App from "./App.jsx";
 import "./index.css";
+import { ErrorBoundary } from "react-error-boundary";
 
 ReactDOM.createRoot(document.getElementById("root")).render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <Provider store={store}>
+ <ErrorBoundary fallback={<h1>Something went wrong</h1>}>
```
Comment (yishain11) [inline client/src/main.jsx:10]: better to add error component, then in the future if we will want to edit/style it - it will be easier

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/main.jsx (hunk 3 above)
Comment (daniahih) [reply client/src/main.jsx]: added ErrorFallBack components

File: client/package.json [hunk 4]
```diff
@@ -10,8 +10,14 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
+
```
Comment (yishain11) [inline client/package.json:14]: remove empty line

File: client/src/App.jsx [hunk 5]
```diff
@@ -1,34 +1,44 @@
-import { useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
-import "./App.css";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import Home from "./Pages/Home";
+import SongPage from "./Pages/SongPage";
+import ArtistPage from "./pages/ArtistPage";
+
```
Comment (yishain11) [inline client/src/App.jsx:5]: remove empty line and commented out code

File: client/src/App.jsx [hunk 6]
```diff
@@ -1,34 +1,44 @@
...
+  },
+  {
+    path: "/Artist",
+    element: <ArtistPage />,
+  },
+]);
 
 function App() {
-  const [count, setCount] = useState(0);
+   // the comment lie is just to see if the error boundary work
```
Comment (yishain11) [inline client/src/App.jsx:24]: remove commented out code

File: client/src/App.jsx [hunk 7]
```diff
@@ -1,34 +1,44 @@
...
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
+        {/* <button onClick={handleClick}>Click me</button> */}
```
Comment (yishain11) [inline client/src/App.jsx:39]: remove commnted out code

File: client/src/main.jsx [hunk 8]
```diff
@@ -1,10 +1,16 @@
...
 
 ReactDOM.createRoot(document.getElementById("root")).render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <Provider store={store}>
+ <ErrorBoundary FallbackComponent={ErrorFallback}>
+  <App />
+</ErrorBoundary>
+
```
Comment (yishain11) [inline client/src/main.jsx:14]: remove empty line

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/App.jsx [hunk 9]
```diff
@@ -1,35 +1,37 @@
...
+    <div>
+      <RouterProvider router={Router} />
       <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
+        <button onClick={handleClick}>Click me</button>
```
Comment (yishain11) [inline client/src/App.jsx:32]: we don't need this button or the handleClick error function. you can remove them

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Pull request merged**